Validation/RecoParticleFlow/plugins/PFJetDQMPostProcessor.cc

```diff
@@ -83,43 +86,85 @@ void PFJetDQMPostProcessor::dqmEndJob(DQMStore::IBooker& ibook_, DQMStore::IGett
+
+    iget_.setCurrentFolder(offsetDir);
+    std::vector<std::string> sME_offset = iget_.getMEs();
+    std::for_each(sME_offset.begin(), sME_offset.end(), [&](auto& s) { s.insert(0, offsetDir); });
 
     iget_.setCurrentFolder(jetResponseDir[idir]);
     std::vector<std::string> sME_response = iget_.getMEs();
     std::for_each(sME_response.begin(), sME_response.end(), [&](auto& s) { s.insert(0, jetResponseDir[idir]); });
-    //for (unsigned int i=0; i<sME_response.size(); i++) std::cout << sME_response[i] << std::endl;
 
     iget_.setCurrentFolder(jetResponseDir[idir]);
 
     double ptBinsArray[ptBins.size()];
     unsigned int nPtBins = ptBins.size() - 1;
     std::copy(ptBins.begin(), ptBins.end(), ptBinsArray);
-    //for(unsigned int ipt = 0; ipt < ptBins.size(); ++ipt) std::cout << ptBins[ipt] << std::endl;
 
     std::string stitle;
     char ctitle[50];
+    std::vector<MonitorElement*> vME_efficiency;
+    std::vector<MonitorElement*> vME_purity;
     std::vector<MonitorElement*> vME_presponse;
+    std::vector<MonitorElement*> vME_presponse_mean;
+    std::vector<MonitorElement*> vME_presponse_median;
     std::vector<MonitorElement*> vME_preso;
     std::vector<MonitorElement*> vME_preso_rms;
+    std::vector<MonitorElement*> vME_ratePUJet;
 
     MonitorElement* me;
     TH1F* h_resp;
-    TH1F* h_genjet_pt;
+    TH1F *h_genjet_pt, *h_genjet_matched_pt;  // *h_genjet_unmatched_pt;
+    TH1F *h_recojet_pt, *h_recojet_matched_pt, *h_recojet_unmatched_pt;
+
+    stitle = offsetDir + "mu";
+    std::vector<std::string>::const_iterator it = std::find(sME_offset.begin(), sME_offset.end(), stitle);
+    if (it == sME_offset.end())
+      continue;
+    me = iget_.get(stitle);
+    int nEvents = ((TH1F*)me->getTH1F())->GetEntries();
+    iget_.setCurrentFolder(jetResponseDir[idir]);
 
     //
     // Response distributions
     //
     for (unsigned int ieta = 1; ieta < etaBins.size(); ++ieta) {
       stitle = genjetDir + "genjet_pt" + "_eta" + seta(etaBins[ieta]);
-      //std::cout << ieta << " " << stitle << std::endl;
 
       std::vector<std::string>::const_iterator it = std::find(sME_genjets.begin(), sME_genjets.end(), stitle);
       if (it == sME_genjets.end())
         continue;
       me = iget_.get(stitle);
       h_genjet_pt = (TH1F*)me->getTH1F();
 
+      if (jetResponseDir[idir].find("noJEC") != std::string::npos) {
+        // getting the histogram for matched gen jets
+        stitle = jetResponseDir[idir] + "genjet_pt" + "_eta" + seta(etaBins[ieta]) + "_matched";
+        me = iget_.get(stitle);
+        h_genjet_matched_pt = (TH1F*)me->getTH1F();
+
+        /*// getting the histogram for unmatched gen jets
+        stitle = jetResponseDir[idir] + "genjet_pt" + "_eta" + seta(etaBins[ieta]) + "_unmatched";
+        me = iget_.get(stitle);
+        h_genjet_unmatched_pt = (TH1F*)me->getTH1F();*/
+      }
+      if (jetResponseDir[idir].find("JEC") != std::string::npos) {
```

Comment:
Every time a "noJEC" is found, as a consequence also a "JEC" is present in the string.
Is this the wanted behaviour? Shouldn't a `else if `be used instead?

Reply:
Done.